src/coreclr/vm/exceptmacros.h

```diff
@@ -308,8 +313,42 @@ VOID DECLSPEC_NORETURN DispatchManagedException(PAL_SEHException& ex, bool isHar
 
 #else // TARGET_UNIX
 
+#ifdef FEATURE_EH_FUNCLETS
+#define INSTALL_MANAGED_EXCEPTION_DISPATCHER                                                \
+    {                                                                                       \
+        auto body = [&]()                                                                   \
```

Comment:
Capturing all locals by reference can cause confusion in many cases. Is this really needed? I appreciate what we are trying to do here and make it generic, but it is very dangerous to "capture everything". This mixing of macro soup _and_ modern C++ capture semantics is likely going to make this very complicated to reason about in crashes.

Reply:
I don't see how we could do it differently. This macro is used at several hundreds of places, so specifying explicit capture list would be a nightmare.
If you have some other idea how to make a piece of code between these macros conditionally run under try / catch or out of that, I'd love to use it instead. I wasn't able to find any other way (other than capturing it in a macro and effectively duplicating the code twice).

Reply:
Is there a perf impact from "capture everything" approach?

Reply:
I was going to run performance benchmarks to see.

Reply:
Is there a reason you couldn't use the try/catch pattern even with the old style eh? While the trick with lambda's is wonderfully clever (and I used in the green threads experiment last year), it carries with it a fairly expensive abstraction penalty that I suspect an extra try/catch does not.

Reply:
I have actually tried that initially, it worked for most of the cases, but there was a problem with `NullReferenceException`. This exception is propagated as SEH exception, so when C++ catch(...) catches that and then rethrows it, it gets converted into access violation (because the internal C++ exception handling code just looks at the exception code and doesn't take the failure address into account). And I couldn't put in an extra SEH __try / __except, because these cannot be combined with C++ EH in the same function (and creating an extra function for the body gets us back to the lambda).

Reply:
I have ran some dotnet/performance microbenchmarks to see the performance impact of the added lambda. First, I was seeing about 15% perf hit in some exception handling tests, but I was able to mitigate it completely by moving the new eh branch stuff to a separate function. 
However, I have realized today that I actually don't need any code here even for the new exception handling. It was not the case until I've added transitioning to the new exception handling in the personality routine, which was done quite late in the development process. Now that the transition is there, it is no longer necessary to catch the exception at places where the `INSTALL_MANAGED_EXCEPTION_DISPATCHER` is used, the personality routine will do the job for us.

Reply:
Removed the INSTALL_MANAGED_EXCEPTION_DISPATCHER  / UNINSTALL_MANAGED_EXCEPTION_DISPATCHER changes for Windows completely, so there is no more lambda / perf impact here.